### PR TITLE
Fix `add_secondary_keys_with_authorization` benchmarks.

### PR DESCRIPTION
**Pull request by Neopallium**
Another TODO cleanup in the identity pallet.

### CODE REVIEW

File: pallets/identity/src/benchmarking.rs [hunk 1]
```diff
@@ -313,46 +314,30 @@ benchmarks! {
...
-    //             auth_signature: sig,
-    //         }
-    //     }).collect::<Vec<_>>();
-
-    // }: _(origin, secondary_keys_with_auth, expires_at)
+    add_secondary_keys_with_authorization {
+        // Number of keys.
+        let i in 0 .. MAX_SECONDARY_KEYS;
+
+        let caller = UserBuilder::<T>::default().generate_did().seed(SEED).build("caller");
```
Comment (Centril) [inline pallets/identity/src/benchmarking.rs:321]: You can use `user` as a shorthand, https://github.com/PolymathNetwork/Polymesh/blob/12abbac83d33ad91f03f1f2ddc1cb3a0bc5aee04/pallets/common/src/benchs/mod.rs#L27

File: pallets/identity/src/benchmarking.rs [hunk 2]
```diff
@@ -313,46 +314,30 @@ benchmarks! {
...
+        let authorization = TargetIdAuthorization::<T::Moment> {
+            target_id: caller.did(),
+            nonce: Module::<T>::offchain_authorization_nonce(caller.did()),
+            expires_at,
+        };
+        let auth_encoded = authorization.encode();
+
+        let mut secondary_keys_with_auth = Vec::with_capacity(i as usize);
+        for x in 0..i {
+            let user = UserBuilder::<T>::default().generate_did().seed(i as u32).build("key");
```
Comment (Centril) [inline pallets/identity/src/benchmarking.rs:333]: Here too

File: pallets/identity/src/benchmarking.rs [hunk 3]
```diff
@@ -313,46 +314,30 @@ benchmarks! {
...
+        let expires_at: T::Moment = 600u32.into();
+        let authorization = TargetIdAuthorization::<T::Moment> {
+            target_id: caller.did(),
+            nonce: Module::<T>::offchain_authorization_nonce(caller.did()),
+            expires_at,
+        };
+        let auth_encoded = authorization.encode();
+
+        let mut secondary_keys_with_auth = Vec::with_capacity(i as usize);
+        for x in 0..i {
```
Comment (Centril) [inline pallets/identity/src/benchmarking.rs:332]: `x` seems unused

File: pallets/identity/src/benchmarking.rs [hunk 4]
```diff
@@ -313,46 +314,30 @@ benchmarks! {
...
+
+        let expires_at: T::Moment = 600u32.into();
+        let authorization = TargetIdAuthorization::<T::Moment> {
+            target_id: caller.did(),
+            nonce: Module::<T>::offchain_authorization_nonce(caller.did()),
+            expires_at,
+        };
+        let auth_encoded = authorization.encode();
+
+        let mut secondary_keys_with_auth = Vec::with_capacity(i as usize);
```
Comment (Centril) [inline pallets/identity/src/benchmarking.rs:331]: Style nit: use `.map(...).collect(...)` instead?

File: pallets/identity/src/benchmarking.rs (hunk 3 above)
Comment (Neopallium) [reply pallets/identity/src/benchmarking.rs:332]: Oh. Good catch.  It should be used as the seed.

**Review by Centril — APPROVED** [review]

**Pull request merged**